Cannot build an app made from wry template

**Describe the bug**
When running `cargo android run` on an app produced freshly and without changes from `wry` template, I receive the following errors:

```
C:\Users\USER\Documents\github\betting_app\app>cargo android run
 (LM-Q630) with target "aarch64-linux-android"
e: C:\Users\USER\Documents\github\betting_app\app\gen\android\app\build.gradle.kts:23:111: Expecting an element
e: C:\Users\USER\Documents\github\betting_app\app\gen\android\app\build.gradle.kts:23:128: Expecting an element
e: C:\Users\USER\Documents\github\betting_app\app\gen\android\app\build.gradle.kts:23:177: Expecting an element
e: C:\Users\USER\Documents\github\betting_app\app\gen\android\app\build.gradle.kts:23:194: Expecting an element
e: C:\Users\USER\Documents\github\betting_app\app\gen\android\app\build.gradle.kts:23:235: Expecting an element
e: C:\Users\USER\Documents\github\betting_app\app\gen\android\app\build.gradle.kts:23:252: Expecting an element
e: C:\Users\USER\Documents\github\betting_app\app\gen\android\app\build.gradle.kts:23:104: Unresolved reference: jniLibs

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\USER\Documents\github\betting_app\app\gen\android\app\build.gradle.kts' line: 23

* What went wrong:
Script compilation errors:

  Line 23:             packaging {                jniLibs.keepDebugSymbols.add("*/arm64-v8a/*.so")                jniLibs.keepDebugSymbols.add("*/armeabi-v7a/*.so")                jniLibs.keepDebugSymbols.add("*/x86/*.so")                jniLibs.keepDebugSymbols.add("*/x86_64/*.so")
                                                                                                                         ^ Expecting an element

  Line 23:             packaging {                jniLibs.keepDebugSymbols.add("*/arm64-v8a/*.so")                jniLibs.keepDebugSymbols.add("*/armeabi-v7a/*.so")                jniLibs.keepDebugSymbols.add("*/x86/*.so")                jniLibs.keepDebugSymbols.add("*/x86_64/*.so")
                                                                                                                                          ^ Expecting an element

  Line 23:             packaging {                jniLibs.keepDebugSymbols.add("*/arm64-v8a/*.so")                jniLibs.keepDebugSymbols.add("*/armeabi-v7a/*.so")                jniLibs.keepDebugSymbols.add("*/x86/*.so")                jniLibs.keepDebugSymbols.add("*/x86_64/*.so")
                                                                                                                                                                                           ^ Expecting an element

  Line 23:             packaging {                jniLibs.keepDebugSymbols.add("*/arm64-v8a/*.so")                jniLibs.keepDebugSymbols.add("*/armeabi-v7a/*.so")                jniLibs.keepDebugSymbols.add("*/x86/*.so")                jniLibs.keepDebugSymbols.add("*/x86_64/*.so")
                                                                                                                                                                                                            ^ Expecting an element

  Line 23:             packaging {                jniLibs.keepDebugSymbols.add("*/arm64-v8a/*.so")                jniLibs.keepDebugSymbols.add("*/armeabi-v7a/*.so")                jniLibs.keepDebugSymbols.add("*/x86/*.so")                jniLibs.keepDebugSymbols.add("*/x86_64/*.so")
                                                                                                                                                                                                                                                     ^ Expecting an element

  Line 23:             packaging {                jniLibs.keepDebugSymbols.add("*/arm64-v8a/*.so")                jniLibs.keepDebugSymbols.add("*/armeabi-v7a/*.so")                jniLibs.keepDebugSymbols.add("*/x86/*.so")                jniLibs.keepDebugSymbols.add("*/x86_64/*.so")
                                                                                                                                                                                                                                                                      ^ Expecting an element

  Line 23:             packaging {                jniLibs.keepDebugSymbols.add("*/arm64-v8a/*.so")                jniLibs.keepDebugSymbols.add("*/armeabi-v7a/*.so")                jniLibs.keepDebugSymbols.add("*/x86/*.so")                jniLibs.keepDebugSymbols.add("*/x86_64/*.so")
                                                                                                                  ^ Unresolved reference: jniLibs

7 errors

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 59s
error: Failed to assemble APK
    command ["C:\\Users\\USER\\Documents\\github\\betting_app\\app\\gen\\android\\gradlew.bat", "--project-dir",
    "C:\\Users\\USER\\Documents\\github\\betting_app\\app\\gen\\android"] exited with code 1
```

**Steps To Reproduce**
Steps to reproduce the behavior. It **must** use tauri-mobile itself and wry template directly instead of tauri-cli.
Create a `wry` template using `tauri-mobile`. Try to run it (on Windows?).

**Expected behavior**
It should build and run successfully

**Screenshots**
N/A

**Platform and Versions (please complete the following information):**
Host OS: `Windows 10 10.0 amd64`
Target OS: ` android arm64       Android 11 (API 30)`
Rustc: `rustc 1.70.0 (90c541806 2023-05-31)`
`cargo mobile doctor`:
```
[✔] cargo-mobile v0.5.3
    • Contains commits up to "feat: add `--skip-targets-install` (#206)\n"
    • Installed at "C:\\Users\\USER\\.cargo\\.tauri-mobile"
    • Windows 10 v10.0
    • rustc v1.70.0 (90c541806 2023-5-31)

[✔] Android developer tools
    • SDK v26.1.1 installed at "C:\\Users\\USER\\AppData\\Local\\Android\\Sdk"
    • NDK v25.0.8775105 installed at "C:\\Users\\USER\\AppData\\Local\\Android\\Sdk\\ndk\\25.0.8775105"

[✔] Connected devices
 (LM-Q630)61
```
(if you build with specific commit, please list it as well)

**Would you want to assign yourself to resolve this bug?**
- [ ] Yes
- [x] No

**Additional context**
Add any other context about the problem here.

--- a/.changes/android-template.md
+++ b/.changes/android-template.md
@@ -1,0 +1,5 @@
+---
+"tauri-mobile": "patch"
+---
+
+Fix Android template generation.

--- a/src/android/project.rs
+++ b/src/android/project.rs
@@ -140,12 +140,15 @@ pub fn gen(
             map.insert("android-app-plugins", metadata.app_plugins());
             map.insert(
                 "android-project-dependencies",
-                metadata.project_dependencies(),
+                metadata.project_dependencies().unwrap_or_default(),
             );
-            map.insert("android-app-dependencies", metadata.app_dependencies());
+            map.insert(
+                "android-app-dependencies",
+                metadata.app_dependencies().unwrap_or_default(),
+            );
             map.insert(
                 "android-app-dependencies-platform",
-                metadata.app_dependencies_platform(),
+                metadata.app_dependencies_platform().unwrap_or_default(),
             );
             map.insert(
                 "has-code",

--- a/templates/platforms/android-studio/app/build.gradle.kts.hbs
+++ b/templates/platforms/android-studio/app/build.gradle.kts.hbs
@@ -1,7 +1,6 @@
 plugins {
     id("com.android.application")
-    id("rust")
-    {{#if android-app-plugins}}{{~#each android-app-plugins}}
+    id("rust"){{#if android-app-plugins}}{{~#each android-app-plugins}}
     id("{{this}}"){{/each}}{{/if}}
 }
 
@@ -26,8 +25,7 @@ android {
             isDebuggable = true
             isJniDebuggable = true
             isMinifyEnabled = false
-            packaging {
-                {{~#each abi-list}}
+            packaging { {{~#each abi-list}}
                 jniLibs.keepDebugSymbols.add("*/{{this}}/*.so"){{/each}}
             }
         }
@@ -49,9 +47,7 @@ rust {
     rootDirRel = "{{root-dir-rel}}"
 }
 
-dependencies {
-    {{#if android-app-dependencies-platform}}{{~#each android-app-dependencies-platform}}
-    implementation(platform("{{this}}")){{/each}}{{/if}}
-    {{#if android-app-dependencies}}{{~#each android-app-dependencies}}
-    implementation("{{this}}"){{/each}}{{/if}}
+dependencies { {{~#each android-app-dependencies-platform}}
+    implementation(platform("{{this}}")){{/each}}{{~#each android-app-dependencies}}
+    implementation("{{this}}"){{/each}}
 }

--- a/templates/platforms/android-studio/build.gradle.kts.hbs
+++ b/templates/platforms/android-studio/build.gradle.kts.hbs
@@ -4,9 +4,8 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath("com.android.tools.build:gradle:8.0.0")
-        {{#if android-project-dependencies}}{{~#each android-project-dependencies}}
-        classpath("{{this}}"){{/each}}{{/if}}
+        classpath("com.android.tools.build:gradle:8.0.0"){{~#each android-project-dependencies}}
+        classpath("{{this}}"){{/each}}
     }
 }
 

